cpu/sam0_common/periph/uart.c

```diff
@@ -317,6 +317,44 @@ void uart_poweroff(uart_t uart)
     sercom_clk_dis(dev(uart));
 }
 
+#ifdef MODULE_PERIPH_UART_COLLISION
+bool uart_collision_detected(uart_t uart)
+{
+    bool collision = dev(uart)->STATUS.bit.COLL;
```

Comment:
The samd5x data sheet states that on collision the first step is to disable TX while the SYNC bit for the CTRLB register is set. Then the collision flag will be set. I'd therefore suggest to wait for the sync bit before polling the collision detection flag:

```C
    /* In case of collision, the CTRLB register
     * will be in sync during disabling of TX,
     * then the flag will be set.
     */
    while (dev(uart)->SYNCBUSY.bit.CTRLB) { }
```